src/store/reducers/app.js

```diff
@@ -1,7 +1,8 @@
-import { SET_ANALYSIS_MODE, CLEAR_ANALYSIS } from '../actions'
+import { SET_ANALYSIS_MODE, CLEAR_ANALYSIS, SET_ANALYSIS_NAME } from '../actions'
 
 const initialState = {
-  analysisMode: null
+  analysisMode: null,
+  viewName: 'Untitled Analysis'
```

Comment:
Similar to the comment about "Untitled Analysis" above -- if we allow an empty string as initial state, it'll be easier to maintain. We only need to display "Untitled Analysis" in the component render, it doesn't need to be stored as a value.